### PR TITLE
refactor(test): use exponential backoff and retry a sample test

### PR DESCRIPTION
**Pull request by AVaksman**
Retry a failing sample test, that causes other dependent tests to fail as well.

Should fix failing samples tests #1182 #1183 #1184 #1185 #1191

### CODE REVIEW

File: samples/system-test/spanner.test.js [hunk 1]
```diff
@@ -66,6 +66,17 @@ const GAX_OPTIONS = {
   },
 };
 
+const delay = async test => {
+  const retries = test.currentRetry();
+  if (retries === 0) return; // no retry on the first failure.
```
Comment (skuruppu) [inline samples/system-test/spanner.test.js:71]:
```suggestion
  // No retry on the first failure.
  if (retries === 0) return;
```

File: samples/system-test/spanner.test.js [hunk 2]
```diff
@@ -66,6 +66,17 @@ const GAX_OPTIONS = {
...
+  // see: https://cloud.google.com/storage/docs/exponential-backoff:
```
Comment (skuruppu) [inline samples/system-test/spanner.test.js:72]:
```suggestion
  // See: https://cloud.google.com/storage/docs/exponential-backoff
```

File: samples/system-test/spanner.test.js [hunk 3]
```diff
@@ -66,6 +66,17 @@ const GAX_OPTIONS = {
...
+  const ms = Math.pow(2, retries) * 250 + Math.random() * 1000;
```
Comment (skuruppu) [inline samples/system-test/spanner.test.js:73]: Out of curiosity, why is it not?
```
const ms = (Math.pow(2, retries) + Math.random()) * 1000;
```

I guess I'm wondering about the 250msec increments of time.

File: samples/system-test/spanner.test.js [hunk 4]
```diff
@@ -293,7 +304,10 @@ describe('Spanner', () => {
   });
 
   // create_storing_index
-  it('should create a storing index in an example table', async () => {
+  it('should create a storing index in an example table', async function () {
+    this.retries(5);
+    await delay(this.test); // delay the start of the test, if this is a retry.
```
Comment (skuruppu) [inline samples/system-test/spanner.test.js:309]:
```suggestion
    // Delay the start of the test, if this is a retry.
    await delay(this.test);
```

**Review by skuruppu — APPROVED** [review]
Just a few minor nits but LGTM otherwise. Thanks heaps for fixing the test issue.

File: samples/system-test/spanner.test.js (hunk 1 above)
Comment (AVaksman) [reply samples/system-test/spanner.test.js]: Done, thanks!

File: samples/system-test/spanner.test.js (hunk 2 above)
Comment (AVaksman) [reply samples/system-test/spanner.test.js]: Done

File: samples/system-test/spanner.test.js (hunk 3 above)
Comment (AVaksman) [reply samples/system-test/spanner.test.js]: Done

File: samples/system-test/spanner.test.js (hunk 4 above)
Comment (AVaksman) [reply samples/system-test/spanner.test.js]: Done

File: samples/system-test/spanner.test.js [hunk 5]
```diff
@@ -293,7 +305,11 @@ describe('Spanner', () => {
   });
 
   // create_storing_index
-  it('should create a storing index in an example table', async () => {
+  it('should create a storing index in an example table', async function () {
+    this.retries(5);
+    // delay the start of the test, if this is a retry.
```
Comment (skuruppu) [inline samples/system-test/spanner.test.js:310]:
```suggestion
    // Delay the start of the test, if this is a retry.
```

**Pull request merged**